src/server/game/Maps/Map.cpp

```diff
@@ -116,6 +116,41 @@ void Map::LoadAllCells()
             LoadGrid((cellX + 0.5f - CENTER_GRID_CELL_ID) * SIZE_OF_GRID_CELL, (cellY + 0.5f - CENTER_GRID_CELL_ID) * SIZE_OF_GRID_CELL);
 }
 
+void Map::RespawnAllObjects()
```

Comment:
is this how Blizzard handles this? as far as I know, there's no respawning at all, players are moved to a different map.

Reply:
yes, thats how blizz handles this, they are not creating a new map. Especially if you think of previous exploits like disabling all doors in Temple of Sethraliss by clearing it first on mythic (0) difficulty

Reply:
its just SMSG_CHANGE_PLAYER_DIFFICULTY_RESULT with LoadingScreenEnable result, not a far teleport